pkg/osquery/tables/windows_associations/windows_associations.go

```diff
@@ -0,0 +1,111 @@
+// +build windows
+
+package windows_associations
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/launcher/pkg/dataflatten"
+	"github.com/kolide/launcher/pkg/osquery/tables/dataflattentable"
+	"github.com/kolide/launcher/pkg/osquery/tables/tablehelpers"
+	"github.com/kolide/osquery-go"
+	"github.com/kolide/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+const dismCmd = "dism.exe"
+
+type Table struct {
+	client *osquery.ExtensionManagerClient
+	logger log.Logger
+}
+
+func TablePlugin(client *osquery.ExtensionManagerClient, logger log.Logger) *table.Plugin {
+
+	columns := dataflattentable.Columns()
+
+	t := &Table{
+		client: client,
+		logger: logger,
+	}
+
+	return table.NewPlugin("kolide_windows_associations", columns, t.generate)
+}
+
+func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var results []map[string]string
+
+	dismResults, err := t.execDism(ctx)
+	if err != nil {
+		level.Info(t.logger).Log("msg", "dism failed", "err", err)
+		return results, err
+	}
+
+	for _, dataQuery := range tablehelpers.GetConstraints(queryContext, "query", tablehelpers.WithDefaults("*")) {
+		flattenOpts := []dataflatten.FlattenOpts{
+			dataflatten.WithQuery(strings.Split(dataQuery, "/")),
+		}
+
+		if t.logger != nil {
+			flattenOpts = append(flattenOpts,
+				dataflatten.WithLogger(level.NewFilter(t.logger, level.AllowInfo())),
+			)
+		}
+
+		rows, err := dataflatten.Xml(dismResults, flattenOpts...)
+		if err != nil {
+			level.Info(t.logger).Log("msg", "flatten failed", "err", err)
+			continue
+		}
+
+		results = append(results, dataflattentable.ToMap(rows, dataQuery, nil)...)
+	}
+
+	return results, nil
+}
+
+func (t *Table) execDism(ctx context.Context) ([]byte, error) {
+	// dism.exe outputs xml, but with weird intermingled status. So
+	// instead, we dump it to a temp file.
```

Comment:
Are the weird 'status headers' omitted from the temp file when the `/Export-DefaultAppAssociations` flag is specified?

Reply:
Yes.

Reply:
Not bouncing through a tmp file would be nice, it feels like pointless diskio.

But, parsing stdout feels fragile (maybe it'd be fine?)

And handing it a socket to write into feels like non-trivial new work.

So this was the simple route